trunk/modules/python/dataset/ingest/labeled.py

```diff
@@ -120,6 +127,11 @@ def readAndDivideData(path, holdoutPercentage, minTest=5, log=None) :
                 log.debug('No files found in [' + root + ']')
             continue
 
+        # find the common file type for the directory
+        # and exclude any files not matching it
+        suffix = mostCommon(files, lambda f: os.path.splitext(f)[1])
+        files = [f for f in files if f.endswith(suffix)]
+
         # add the new label for the current directory

```

Comment:
We should optionally perform these checks as these will be expensive and most of our datasets will be uniformly sized.